### ISSUE TITLE
setting SECRETS SCOPE by ENDPOINT?

### ISSUE THREAD

**Issue by cboettig** [duckdb/duckdb-httpfs#5]
duckdb documentation describes how to create [multiple secrets for the same service](https://duckdb.org/docs/configuration/secrets_manager), e.g. if we have two S3 buckets that need different credentials.  The directions indicating scoping based on _bucket name_.  

However, I commonly encounter the case where I need to access two different S3 services at different _endpoints_ -- for instance, S3 buckets data.source.coop , S3 buckets on a local minio instance,  S3 buckets on AWS.  It would be natural to scope these by the _endpoint name_, not the bucket name -- I may have the same credentials for multiple buckets, and I may have the same bucket name on different endpoints.  

I think it is something of a bug that scope focuses on bucket name but ignores endpoint.  Would you consider adding support for this?

**Comment by Alex-DVL** [comment]
I was guessing that you can combine  ENDPOINT and SCOPE  parameters while creating a secret , even though scope this is not (yet ) documented  , did you try that ?

**Comment by joebowbeer** [comment]
You can combine ENDPOINT and SCOPE in a S3 SECRET.

SCOPE (currently) is specified as an s3://prefix

Examples of how SCOPE is used to select a secret:

```sql
FROM glob('s3://prefix/*');
```

```sql
ATTACH 'ducklake:metadata.ducklake' AS my_ducklake (DATA_PATH 's3://prefix');
```

The endpoint does not make an appearance in these examples, so (currently) there's no way to use endpoint to select a secret.

I think you're forced to differentiate on the bucket/path prefix.

**Comment by cboettig** [comment]
Yes, precisely. this works, but is not intuitive and sometimes creates issues because this constraint does not occur in almost any other implementation of S3 authentication (arrow, GDAL, minio boto3 / official sdk and all derivatives).  

For instance, we occasionally have needs to migrate between S3 endpoints (official AWS vs National Science Foundation's CEPH vs local MINIO).  Workflows for all other client libraries handle this pretty seamlessly as we can just change an endpoint environmental variable, but bucket names and paths are often hardwired into scripts and code (while endpoint URLs almost never are).  This means that changing the bucket name when mirroring breaks the code.  I know this is surely seen as an edge case, but it still seems odd for duckdb to be handling this in a very different way than most other libraries.  

but duckdb rocks so meanwhile we're just making sure we have unique bucket names so we can use this workaround!
